pkg/test/framework/components/cluster/fake.go

```diff
@@ -0,0 +1,36 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cluster
+
+import (
+	"istio.io/istio/pkg/kube"
+)
+
+var _ Cluster = FakeCluster{}
+
+func init() {
+	RegisterFactory(Fake, newFakeCluster)
+}
+
+func newFakeCluster(cfg Config, topology Topology) (Cluster, error) {
+	return &FakeCluster{ExtendedClient: kube.MockClient{}, Topology: topology}, nil
+}
+
+// FakeCluster used for testing.
+type FakeCluster struct {
```

Comment:
nit: Could just call this `Fake`, since it would be `cluster.Fake`.

Reply:
That would mean renaming all of the cluster kinds to have a suffix, since we already have `Fake Kind = "Fake"`.